features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/RumFeature.kt

```diff
@@ -393,7 +391,7 @@ internal class RumFeature(
 
     private fun initializeVitalReaders(periodInMs: Long) {
         @Suppress("UnsafeThirdPartyFunctionCall") // pool size can't be <= 0
-        vitalExecutorService = LoggingScheduledThreadPoolExecutor(1, sdkCore.internalLogger)
+        vitalExecutorService = sdkCore.createScheduledExecutorService()
```

Comment:
maybe the possible improvement for the future: `LoggingScheduledThreadPoolExecutor` is calling `ScheduledThreadPoolExecutor` constructor which allow unlimited growth of the number of threads in the pool (it is limited by `Int.MAX_VALUE`).

Unlikely we may have a big number of threads there, but if by some reason operations scheduled on this executor are very slow, in order to satisfy the execution schedule we may see uncontrolled growth of the number of threads in the pool. Probably we can limit max value.

Reply:
For now all our scheduledExecutor instances should only have  one queued task at a time, which mean we shouldn't need more than one thread each.

Reply:
That is true for most of our use-cases, except vitals: there we are scheduling 2 different tasks - CPU and Memory read. But anyway, my comment was rather for the generic use-case.